### PR TITLE
UCP/TEST: Fix rkey_ptr protocol and add stats/tests

### PR DESCRIPTION
**Pull request by yosefe**
# Why
Rendezvous over rkey_ptr (xpmem) is broken: 
1. xpmem rkey is not sent because of missing bit in rma_bw_md_map
2. few memory access issues in rkey_ptr progress function

# How
Fix issues 1,2 and add statistics and unit tests to make sure rkey_ptr is always used in the presence of xpmem

### CODE REVIEW

**Comment by yosefe** [comment]
@brminich @dmitrygx can you pls take a look?

File: src/ucp/tag/rndv.c [hunk 1]
```diff
@@ -222,6 +222,7 @@ ucs_status_t ucp_tag_send_start_rndv(ucp_request_t *sreq)
         ucs_assert(sreq->send.lane == ucp_ep_get_am_lane(ep));
         sreq->send.uct.func = ucp_proto_progress_rndv_rts;
         status              = ucp_tag_rndv_reg_send_buffer(sreq);
+
```
Comment (dmitrygx) [inline src/ucp/tag/rndv.c:225]: Why needed?

Comment (yosefe) [reply src/ucp/tag/rndv.c:225]: leftover..

File: src/ucp/tag/rndv.c [hunk 2]
```diff
@@ -512,6 +512,8 @@ UCS_PROFILE_FUNC(ucs_status_t, ucp_rndv_progress_rma_get_zcopy, (self),
         ucp_rkey_destroy(rndv_req->send.rndv_get.rkey);
         ucp_rndv_recv_data_init(rndv_req->send.rndv_get.rreq,
                                 rndv_req->send.length);
+        UCP_WORKER_STAT_RNDV(ep->worker, GET_ZCOPY, -1);
+        UCP_WORKER_STAT_RNDV(ep->worker, RTR,       +1);
```
Comment (brminich) [inline src/ucp/tag/rndv.c:516]: we increase `RX_RNDV_RTR` counter before to **send** RTR. Looks confusing,
also there are other places where we send rtr, maybe increase stats in `ucp_rndv_req_send_rtr`?

File: test/gtest/ucp/test_ucp_tag_xfer.cc [hunk 3]
```diff
@@ -1050,14 +1051,54 @@ class test_ucp_tag_stats : public test_ucp_tag_xfer {
...
+
+        if (has_xpmem()) {
+            /* rkey_ptr expected to be selected if xpmem is available */
+            EXPECT_EQ(1u, rkey_ptr);
+        } else if (has_get_zcopy()) {
+            /* if any transprots supports get_zcopy, expect it to be used */
+            EXPECT_EQ(1u, get_zcopy);
+        } else {
+            /* Could be a transport which supports get_zcopy that wasn't
+             * accounted for, or fallback to RTR. In any case, rpey_ptr is not
```
Comment (brminich) [inline test/gtest/ucp/test_ucp_tag_xfer.cc:1095]: `rkey_ptr`

File: test/gtest/ucp/test_ucp_tag_xfer.cc [hunk 4]
```diff
@@ -1050,14 +1051,54 @@ class test_ucp_tag_stats : public test_ucp_tag_xfer {
...
+        UCS_TEST_MESSAGE << "get_zcopy: " << get_zcopy
+                         << " rtr: " << rtr
+                         << " rkey_ptr: " << rkey_ptr;
+        EXPECT_EQ(1, get_zcopy + rtr + rkey_ptr);
+
+        if (has_xpmem()) {
+            /* rkey_ptr expected to be selected if xpmem is available */
+            EXPECT_EQ(1u, rkey_ptr);
+        } else if (has_get_zcopy()) {
+            /* if any transprots supports get_zcopy, expect it to be used */
```
Comment (brminich) [inline test/gtest/ucp/test_ucp_tag_xfer.cc:1091]: `transports`

File: src/ucp/tag/rndv.c [hunk 5]
```diff
@@ -1164,6 +1165,7 @@ UCS_PROFILE_FUNC_VOID(ucp_rndv_matched, (worker, rreq, rndv_rts_hdr),
      * configured to PUT, or GET rndv mode is unsupported - send an RTR and
      * the sender will send the data with active message or put_zcopy. */
     ucp_rndv_recv_data_init(rreq, rndv_rts_hdr->size);
+    UCP_WORKER_STAT_RNDV(ep->worker, RTR, +1);
```
Comment (brminich) [inline src/ucp/tag/rndv.c:1168]: maybe just `1`?

File: src/ucp/tag/rndv.c (hunk 2 above)
Comment (yosefe) [reply src/ucp/tag/rndv.c:516]: `ucp_rndv_req_send_rtr` is also used from fragment-rtr and we don't want to count those

Comment (brminich) [reply src/ucp/tag/rndv.c:517]: what is about naming? RTR counter reflects sending of `RTR`, but it's name is `RX_RNDV_RTR`

**Pull request merged**